std/exception.d

```diff
@@ -1196,32 +1196,40 @@ if (__traits(isRef, source) || isDynamicArray!S ||
     {
         static if (is(S == void[n], size_t n))
         {
-            static if (n >= (void*).sizeof)
+            static if (n >= (void[]).sizeof)
             {
-                // Reinterpreting cast is impossible during ctfe
-                if (__ctfe)
-                    return true;
-
-                // Only check for properly aligned pointers
-                enum al = (void*).alignof - 1;
-                const base = cast(size_t) &source;
-                const alBase = (base + al) & ~al;
-                const elems = (n - (alBase - base)) / (void*).sizeof;
-
-                foreach (const s; (cast(void**) alBase)[0 .. elems])
+                // could contain a slice, which could point at anything.
+                return true;
+            }
+            else
+            {
+                static if (n >= (void*).sizeof)
                 {
-                    if (mayPointTo(s, target))
+                    // Reinterpreting cast is impossible during ctfe
+                    if (__ctfe)
                         return true;
+
+                    // Only check for properly aligned pointers
+                    enum al = (void*).alignof - 1;
+                    const base = cast(size_t) &source;
+                    const alBase = (base + al) & ~al;
+                    const elems = (n - (alBase - base)) / (void*).sizeof;
+
+                    foreach (const s; (cast(void**) alBase)[0 .. elems])
+                    {
+                        if (mayPointTo(s, target))
+                            return true;
```

Comment:
This path could be simplified because the chunk can contain at most one aligned pointer, e.g.:

```
if ((n - (alBase - base)) >= (void*).sizeof && mayPointTo(*alBase, target))
    return true;
```